sycl/source/backend/opencl.cpp

```diff
@@ -44,6 +45,74 @@ __SYCL_EXPORT queue make_queue(const context &Context,
   return detail::make_queue(NativeHandle, Context, nullptr, false,
                             ContextImpl->get_async_handler(), backend::opencl);
 }
+
+// Free functions to query OpenCL backend extensions
+// TODO: Extensions have been deprecated for aspects
+__SYCL_EXPORT bool has_extension(const sycl::platform &SyclPlatform,
+                                 const std::string &Extension) {
+  if (SyclPlatform.get_backend() != sycl::backend::opencl) {
+    throw sycl::exception(
+        errc::backend_mismatch,
+        "has_extension can only be used with an OpenCL backend");
+  }
+
+  auto PlatformImpl = getSyclObjImpl(SyclPlatform);
+  RT::PiPlatform PluginPlatform = PlatformImpl->getHandleRef();
+  plugin Plugin = PlatformImpl->getPlugin();
+
+  // Manual invocation of plugin API to avoid using deprecated
+  // info::platform::extensions call
+  size_t ResultSize = 0;
+  Plugin.call<PiApiKind::piPlatformGetInfo>(
+      PluginPlatform, PI_PLATFORM_INFO_EXTENSIONS, 0, nullptr, &ResultSize);
+  if (ResultSize == 0) {
+    return false;
+  }
+  std::unique_ptr<char[]> Result(new char[ResultSize]);
```

Comment:
I don't think so, but we can use `std::string_view ExtensionsString` to avoid copy and still get the `operator==` for use in `std::find`.

Reply:
>> I don't think so,

Could you please clarify why?

>> but we can use std::string_view ExtensionsString to avoid

Isn't `std::string_view` is a non-owning container?

Reply:
Even if std::string would work, that would be very non-conventional use of filling it by a C-style API by providing that API with a raw pointer into it which I don't consider a good solution. I'd very much prefer the current approach of having a smart pointer for the memory allocation instead.

> Isn't std::string_view a non-owning container? 

Yes, that's why we'll still need this `Result` smart-pointer to own the memory. But we'd be able to use the view for subsequent manipulations without creating copies.

Reply:
I used the logic from the existing platform/device info retrieval methods as I thought it would be better to keep it as self-contained as possible and thought that process was standard procedure. 

- Device: https://github.com/intel/llvm/blob/sycl/sycl/source/detail/device_info.hpp#L160
- Platform: https://github.com/intel/llvm/blob/sycl/sycl/source/detail/platform_info.hpp#L21

What are your recommendations? Keep as-is, use the above inline helpers, or rewrite the existing with string_view?

Reply:
`string_view`